pkg/tsdb/prometheus/prometheus.go

```diff
@@ -62,17 +70,23 @@ func init() {
 }
 
 func (e *PrometheusExecutor) getClient(dsInfo *models.DataSource) (apiv1.API, error) {
-	cfg := api.Config{
-		Address:      dsInfo.Url,
-		RoundTripper: e.Transport,
+	// Would make sense to cache this but executor is recreated on every alert request anyway.
```

Comment:
@aknuds1 not sure if by design or bug. Seems like in https://github.com/grafana/grafana/blob/d7862c50b8bf8d5351cd9110b4bcd09283cb1bf7/pkg/tsdb/service.go#L73 we should be adding the created plugin to the PluginManager otherwise plugin is nil on every request.